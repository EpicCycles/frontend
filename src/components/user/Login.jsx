import React from 'react';
import { Dimmer, Loader } from 'semantic-ui-react';
import { PASSWORD, TEXT } from '../app/model/helpers/fields';
import { VALUE_MISSING } from '../app/model/helpers/error';
import { updateModel } from '../app/model/helpers/model';
import { updateObject } from '../../helpers/utils';
import EditModelPage from '../app/model/EditModelPage';
import IconArray from '../../common/IconArray';
const loginFields = [
  {
    fieldName: 'username',
    type: TEXT,
    displaySize: 20,
    maxLength: 60,
    header: 'User Name',
    synonyms: [],
    required: true,
    error: VALUE_MISSING,
  },
  {
    fieldName: PASSWORD,
    type: PASSWORD,
    displaySize: 20,
    maxLength: 60,
    header: 'Password',
    synonyms: [],
    required: true,
    error: VALUE_MISSING,
  },
];
const initialUser = { username: '', password: '' };
class Login extends React.Component {
  state = {
    userDetail: updateObject(initialUser),
  };
  handleModelValueChange = (fieldName, input) => {
    let { userDetail } = this.state;

    userDetail = updateModel(userDetail, loginFields, fieldName, input);
    this.setState({ userDetail });
  };

  onClickReset = () => {
    const userDetail = updateObject(initialUser);
    this.setState({ userDetail });
  };
  loginUser = () => {
    const { userDetail } = this.state;
    this.props.loginUser(userDetail.username, userDetail.password);
  };
  handleKeyPress = e => {
    if (e.key === 'Enter') {
      const { userDetail } = this.state;
      if (userDetail.username && userDetail.password) this.loginUser();
    }
  };

  render() {
    const { userDetail } = this.state;
    const { isLoading } = this.props;
    const additionalActions = [
      {
        iconName: 'sign in',
        iconTitle: 'Sign In',
        iconAction: () => this.loginUser(),
        iconDisabled: !(userDetail.username && userDetail.password),
      },
    ];
    return (
      <form onKeyPress={this.handleKeyPress}>
        <div className="fit-content">
          <h1>Login</h1>
          <EditModelPage
            model={userDetail}
            modelFields={loginFields}
            onChange={this.handleModelValueChange}
          />
          <div className="full align_right">
            <IconArray componentKey={''} actionArray={additionalActions} />
          </div>
          {isLoading && (
            <Dimmer active inverted>
              <Loader content="Loading" />
            </Dimmer>
          )}
        </div>
      </form>
    );
  }
}

export default Login;
