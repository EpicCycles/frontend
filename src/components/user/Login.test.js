import React from 'react';
import toJson from 'enzyme-to-json';
import Login from './Login';

describe('Login', () => {
  it('displays login fields when user in session', () => {
    const component = shallow(<Login user={{ username: 'testUser' }} s />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('displays login fields when no user in session', () => {
    const component = shallow(<Login />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
