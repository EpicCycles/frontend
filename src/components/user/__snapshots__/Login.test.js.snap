// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login displays login fields when no user in session 1`] = `
<form
  onKeyPress={[Function]}
>
  <div
    className="fit-content"
  >
    <h1>
      Login
    </h1>
    <EditModelPage
      className="fit-content"
      model={
        Object {
          "password": "",
          "username": "",
        }
      }
      modelFields={
        Array [
          Object {
            "displaySize": 20,
            "error": "A value is required.",
            "fieldName": "username",
            "header": "User Name",
            "maxLength": 60,
            "required": true,
            "synonyms": Array [],
            "type": "text",
          },
          Object {
            "displaySize": 20,
            "error": "A value is required.",
            "fieldName": "password",
            "header": "Password",
            "maxLength": 60,
            "required": true,
            "synonyms": Array [],
            "type": "password",
          },
        ]
      }
      onChange={[Function]}
    />
    <div
      className="full align_right"
    >
      <IconArray
        actionArray={
          Array [
            Object {
              "iconAction": [Function],
              "iconDisabled": true,
              "iconName": "sign in",
              "iconTitle": "Sign In",
            },
          ]
        }
        componentKey=""
      />
    </div>
  </div>
</form>
`;

exports[`Login displays login fields when user in session 1`] = `
<form
  onKeyPress={[Function]}
>
  <div
    className="fit-content"
  >
    <h1>
      Login
    </h1>
    <EditModelPage
      className="fit-content"
      model={
        Object {
          "password": "",
          "username": "",
        }
      }
      modelFields={
        Array [
          Object {
            "displaySize": 20,
            "error": "A value is required.",
            "fieldName": "username",
            "header": "User Name",
            "maxLength": 60,
            "required": true,
            "synonyms": Array [],
            "type": "text",
          },
          Object {
            "displaySize": 20,
            "error": "A value is required.",
            "fieldName": "password",
            "header": "Password",
            "maxLength": 60,
            "required": true,
            "synonyms": Array [],
            "type": "password",
          },
        ]
      }
      onChange={[Function]}
    />
    <div
      className="full align_right"
    >
      <IconArray
        actionArray={
          Array [
            Object {
              "iconAction": [Function],
              "iconDisabled": true,
              "iconName": "sign in",
              "iconTitle": "Sign In",
            },
          ]
        }
        componentKey=""
      />
    </div>
  </div>
</form>
`;
